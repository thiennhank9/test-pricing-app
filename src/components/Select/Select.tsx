import React from "react";
import "./Select.css";

interface SelectProps {
  value: string;
  options: Array<string>;
  onChange(event: React.ChangeEvent<HTMLSelectElement>): void;
}

function Select(props: SelectProps) {
  const { value, options, onChange } = props;

  return (
    <select onChange={onChange} value={value} className="select">
      {options.map((option) => (
        <option key={option} value={option}>
          {option}
        </option>
      ))}
    </select>
  );
}

export default Select;
