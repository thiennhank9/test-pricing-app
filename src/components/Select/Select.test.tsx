import TestRenderer from "react-test-renderer";
import Select from "./Select";

describe("Select", () => {
  test("render select", () => {
    const testRenderer = TestRenderer.create(<Select value="1" options={["1", "2", "3"]} onChange={jest.fn()}/>);
    const testInstance = testRenderer.root;
    expect(testRenderer.toJSON()).toMatchSnapshot();
    const options = testInstance.findAllByType('option');
    expect(options.length).toBe(3);
  });
});
