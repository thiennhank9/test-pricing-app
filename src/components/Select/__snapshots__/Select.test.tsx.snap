// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select render select 1`] = `
<select
  className="select"
  onChange={[MockFunction]}
  value="1"
>
  <option
    value="1"
  >
    1
  </option>
  <option
    value="2"
  >
    2
  </option>
  <option
    value="3"
  >
    3
  </option>
</select>
`;
